uqcsbot/text.py

```diff
@@ -28,33 +29,33 @@ def __init__(self, bot: commands.Bot):
         self.bot.tree.add_command(self.scare_menu)
 
     @app_commands.command()
-    @app_commands.describe(message="Input string")
-    async def binify(self, interaction: discord.Interaction, message: str):
+    @app_commands.describe(message="Input string", encoding="Character encoding to use, defaults to UTF-8")
+    async def binify(self, interaction: discord.Interaction, message: str, encoding: Optional[str] = "utf-8"):
```

Comment:
Making encoding a `Optional[Literal["utf-8", "ascii"]]` would allow discord to give auto-suggestions for the encoding. `Literal["utf-8", "ascii"]` could also be made its own type.

Reply:
You would want to list all the options from what I've read [here](https://discordpy.readthedocs.io/en/stable/ext/commands/commands.html).

Reply:
That would mean making a `Literal[]` listing every single codec on the page...
Or choosing just a subset of them to list. I don't like either solution particularly much, but I'm happy to do it if it's preferable to having zero autosuggestions.

Reply:
I now *have* looked into this. I think a limited set of autosuggestions is the best case here - my dodgy union type idea doesn't work, and in any case, allowing arbitrary str input can throw `LookupError: unknown encoding`, which breaks the bot.

I reckon just pick your favourite 5-10 English encodings and add them to a literal type.

Reply:
How nice are you imagining the errors? Would a simple link to the list page be ok?

> "Invalid encoding. A list of valid encodings can be found at https://docs.python.org/3/library/codecs.html#standard-encodings"

Otherwise a rough autosuggestion might be good, but too complex to add in this PR.

Reply:
I'm happy with basically anything that's not "The application did not respond" tbh; a link sounds good to me.

Reply:
It looks like autocomplete suggestions can be done with a function decorator as well: https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.app_commands.autocomplete
Doing that still allows arbitrary string input

Reply:
TIL, and if you want to whack one of them on, be my guest

Reply:
Done!

Reply:
Looks great!